### PR TITLE
Ab#110903 parameterizing pipeline deploy

### PR DESCRIPTION
**Pull request by raviyerramilli**
Moved tenant_id = sys.argv[3]  #P3 above 

resp = requests.get('[https://login.microsoftonline.com/{}/oauth2/token](https://login.microsoftonline.com/%7B%7D/oauth2/token)'.format(tenant_id), data=data)

To resolve error: UnboundLocalError: local variable 'tenant_id' referenced before assignment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,39 +14,72 @@ PBIX file should be placed in a folder which corresponds to the key in the deplo
...
+        uses: nathangiusti/Power-BI-Pipeline-Deploy@v2.1
+        with:
+          files: ${{ steps.changed-files.outputs.all_modified_files }}
+        env:
+          CLIENT_ID: ${{ secrets.NON_PROD_SPN_ID }}
+          CLIENT_SECRET: ${{ secrets.NON_PROD_SPN_SECRET }}
+
+Following is the sample action.yaml file needed to configure the action itself
+
+name: 'Power BI Pipeline Deploy'
```
Comment (nathangiusti) [inline README.md]: Take this sample action.yaml out. 

Think about the difference between the action.yaml file and the workflow yaml file. The action yaml.file describes the action. The workflow yaml file is what is what people actually create to use the action. We want to include a sample workflow.

File: scripts/pipeline_deploy.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
+from ast import arg
```
Comment (nathangiusti) [inline scripts/pipeline_deploy.py]: Do we need this?

File: scripts/pipeline_deploy.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 import os
 import sys
 from pathlib import Path
 
 import requests
 import yaml
+import argparse
```
Comment (nathangiusti) [inline scripts/pipeline_deploy.py]: You can comment this out if you wanna take shot at the arg parse stuff later.

File: scripts/pipeline_deploy.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 
 #tenant_id = 'cef04b19-7776-4a94-b89b-375c77a8f936'
```
Comment (nathangiusti) [inline scripts/pipeline_deploy.py]: Go ahead delete these commented lines.

**Review by nathangiusti — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (raviyerramilli) [reply README.md]: Got it so, I'll the actions.yaml out and leave the Deploy to prod workflow sample part in there as it is very similar to my Deploy to test.yaml

File: scripts/pipeline_deploy.py (hunk 3 above)
Comment (raviyerramilli) [reply scripts/pipeline_deploy.py]: Yup, planning to use argparse in my next stint.

File: scripts/pipeline_deploy.py (hunk 2 above)
Comment (raviyerramilli) [reply scripts/pipeline_deploy.py]: don't think so as they doesn't seem to be any parse() used in the code according this doc:
https://www.educative.io/edpresso/what-is-astargarg-annotation-typecomment-in-python

File: scripts/pipeline_deploy.py [hunk 5]
```diff
@@ -21,14 +20,7 @@ def parse_response(response):
...
 
-    #parser = argparse.ArgumentParser()
-    #parser.add_argument("--files", nargs=1)
-    #parser.add_argument("--db_url", )
-    #parser.add_argument("--seperator", nargs=1)
-    #parser.add_argument("--tenant_id", nargs=1)
-    #parser.add_argument("--source-stage-order", nargs=1)
-    #parser.add_argument("--update-app-in-targetWorkspace", nargs=1)
-    #parser.add_argument("--seperator", nargs=1)
+
```
Comment (nathangiusti) [inline scripts/pipeline_deploy.py]: remove the extra white space as well.

**Review by nathangiusti — CHANGES_REQUESTED** [review]

**Review by nathangiusti — APPROVED** [review]

**Pull request merged**